x-pack/plugins/security_solution/public/detections/components/host_isolation/use_host_isolation_action.tsx

```diff
@@ -193,6 +219,16 @@ export const useHostIsolationAction = ({
       return menuItems;
     }
 
+    if (
+      crowdstrikeAgentId &&
+      isCrowdstrikeAlert &&
+      crowdstrikeManualHostActionsEnabled &&
+      hasActionsAllPrivileges &&
+      (canIsolateHost || (isHostIsolated && !canUnIsolateHost))
+    ) {
+      return menuItems;
+    }
```

Comment:
I'm curious why this logic is not similar to sentinel one's (above this if block)? Why are we doing additional check for `(canIsolateHost || (isHostIsolated && !canUnIsolateHost))`

Reply:
Good catch!